executor/write.go

```diff
@@ -912,7 +912,7 @@ func (e *InsertValues) initDefaultValues(row []types.Datum, marked map[int]struc
 				return errors.Trace(err)
 			}
 			row[i].SetInt64(val)
-			if val != 0 {
+			if val != 0 || (e.ctx.GetSessionVars().SQLMode&mysql.ModeNoAutoValueOnZero) > 0 {
```

Comment:
If I understood correctly, this scenario is already covered in the test case https://github.com/pingcap/tidb/pull/3661/files#diff-27e6281ea7526ea83fff97f6883294b4R288